packages/@aws-cdk-testing/framework-integ/test/aws-autoscaling-hooktargets/test/integ.queue-hook.js.snapshot/integ-autoscalinghook-queue.template.json

```diff
@@ -451,39 +451,85 @@
     ]
    }
   },
-  "GroupLaunchConfigEA6271FF": {
-   "Type": "AWS::AutoScaling::LaunchConfiguration",
+  "GroupLaunchTemplate9C55CCE4": {
+   "Type": "AWS::EC2::LaunchTemplate",
    "Properties": {
-    "ImageId": {
-     "Ref": "al2022AMIParameter"
-    },
-    "InstanceType": "t3.small",
-    "IamInstanceProfile": {
-     "Ref": "GroupInstanceProfileA2F3B693"
+    "LaunchTemplateData": {
+     "IamInstanceProfile": {
+      "Arn": {
+       "Fn::GetAtt": [
+        "GroupLaunchTemplateProfile087E2517",
+        "Arn"
+       ]
+      }
+     },
+     "ImageId": {
+      "Ref": "al2023AMIParameter"
+     },
+     "InstanceType": "t3.small",
+     "Monitoring": {
+      "Enabled": false
+     },
+     "SecurityGroupIds": [
+      {
+       "Fn::GetAtt": [
+        "GroupInstanceSecurityGroupBAF83E2C",
+        "GroupId"
+       ]
+      }
+     ],
+     "TagSpecifications": [
+      {
+       "ResourceType": "instance",
+       "Tags": [
+        {
+         "Key": "Name",
+         "Value": "integ-autoscalinghook-queue/Group/LaunchTemplate"
+        }
+       ]
+      },
+      {
+       "ResourceType": "volume",
+       "Tags": [
+        {
+         "Key": "Name",
+         "Value": "integ-autoscalinghook-queue/Group/LaunchTemplate"
+        }
+       ]
+      }
+     ],
+     "UserData": {
+      "Fn::Base64": "#!/bin/bash"
+     }
     },
-    "SecurityGroups": [
+    "TagSpecifications": [
      {
-      "Fn::GetAtt": [
-       "GroupInstanceSecurityGroupBAF83E2C",
-       "GroupId"
+      "ResourceType": "launch-template",
+      "Tags": [
+       {
+        "Key": "Name",
+        "Value": "integ-autoscalinghook-queue/Group/LaunchTemplate"
+       }
       ]
      }
-    ],
-    "UserData": {
-     "Fn::Base64": "#!/bin/bash"
-    }
-   },
-   "DependsOn": [
```

Comment:
It's losing the dependency on the `Role`. It still has the implicit dependency on the instance profile because of the reference, but there needs to be an explicit dependency on the role.

Reply:
Good catch. I've added a launch template dependency on the role.